testing/hooks_test.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+import { afterEach, beforeEach, withHooks } from "./hooks.ts";
+import { assertEquals } from "./asserts.ts";
+
+Deno.test("beforeEach is exectued correctly", function (): void {
+  let didInvoke = false;
+  beforeEach(function () {
+    didInvoke = true;
+    assertEquals(/deno/, /deno/);
+  });
+  assertEquals(didInvoke, false);
+  withHooks(function () {
+    assertEquals(didInvoke, true);
+  })();
+});
+
+Deno.test("afterEach is exectued correctly", function (): void {
+  let didInvoke = false;
+  afterEach(function () {
+    didInvoke = true;
+    assertEquals(didInvoke, true);
+  });
+  assertEquals(didInvoke, false);
+  withHooks(function () {
+    assertEquals(/cat/, /cat/);
+  })();
```

Comment:
I think we should check didInvoke === false inside the main function, and check didInvoke === true after the main, to ensure the the hook is actually called after the main func. It might look something like the below

```suggestion
  afterEach(function () {
    didInvoke = true;
  });
  assertEquals(didInvoke, false);
  withHooks(function () {
    assertEquals(didInvoke, false);
  })();
  assertEquals(didInvoke, true);
```

Reply:
It will return false unless we remove `async` and `await` from `withHooks` or adding await before 
```ts
await withHooks(function () {   
 assertEquals(didInvoke, false);  })();
}
```

Reply:
> adding await before withHook

This sounds good to me. BTW I think we should check `assertEquals(didInvoke, false)` inside the main function to ensure it's actually called after it.

Reply:
yes, it's good solution to check that